### ISSUE TITLE
LightmapGI: Baking lightmap in medium quality or more generate green invalid probes (low generate black invalid probes)

### ISSUE THREAD

**Issue by jcostello** [godotengine/godot#77169]
### Godot version

v4.1.dev.custom_build [437041a98]

### System information

Ubuntu with Nvidia 2060

### Issue description

IDK the reason why but in medium quality or bigger, invalid probes are green. They  should not be taking into account

Medium Quality

![Screenshot from 2023-05-17 15-22-34](https://github.com/godotengine/godot/assets/1168582/1cb65f50-f82b-4386-b4df-0761e3e02e3b)

Low Quality
![Screenshot from 2023-05-17 15-16-32](https://github.com/godotengine/godot/assets/1168582/b937c390-43c2-46ef-874b-067cb5da3cb3)

### Steps to reproduce

Use MRP

Bake the lightmap (in medium quality)
Move the mesh in the editor to get the dynamic lighting.
You will get same result as the picture above

### Minimal reproduction project

[lightmap.zip](https://github.com/godotengine/godot/files/11501429/lightmap.zip)

**Comment by gongpha** [comment]
I can't reproduce it in 2e273f0e35800f2bcc7c05db54100caa97928f69, whether the medium, high or ultra quality was used
( Windows 10, NVIDIA GeForce GTX 1650 )
![image](https://github.com/godotengine/godot/assets/13400398/cff7339d-2171-49bc-9b48-6c1456faf1c1)

**Comment by jcostello** [comment]
I updated the MRP. The issue still happens. Check it out @gongpha 

[Lightmap2.zip](https://github.com/godotengine/godot/files/11925507/Lightmap2.zip)

![Screenshot from 2023-07-01 07-02-24](https://github.com/godotengine/godot/assets/1168582/6f801024-52d2-4514-bb47-6d4854a8b2af)

**Comment by Calinou** [comment]
I can't open Lightmap2.zip's main scene in the editor:

```
  scene/resources/resource_format_text.cpp:283 - res://Level.tscn:276 - Parse Error: 
  Failed loading resource: res://Level.tscn. Make sure resources have been imported by opening the project in the editor at least once.
  editor/editor_data.cpp:626 - Index p_idx = 1 is out of bounds (edited_scene.size() = 1).
  scene/resources/resource_format_text.cpp:447 - res://Materials/Prototype.tres:3 - ext_resource, invalid UID: uid://e46ya640nfhb - using text path instead: res://Textures/prototype_textures/Light/texture_07.png
  Resource file not found: res://Textures/prototype_textures/Light/texture_07.png.
  scene/resources/resource_format_text.cpp:447 - res://Level.tscn:7 - ext_resource, invalid UID: uid://cb7h7egu8wg3e - using text path instead: res://Level.lmbake
  Cannot open file 'res://Level.lmbake'.
  Failed loading resource: res://Level.lmbake. Make sure resources have been imported by opening the project in the editor at least once.
  scene/resources/resource_format_text.cpp:283 - res://Level.tscn:276 - Parse Error: 
  Failed loading resource: res://Level.tscn. Make sure resources have been imported by opening the project in the editor at least once.
  editor/editor_data.cpp:626 - Index p_idx = 1 is out of bounds (edited_scene.size() = 1).
```

**Comment by jcostello** [comment]
@Calinou try now.

I didnt get the 
```
No meshes to bake. Make sure they contain an UV2 channel and that the 'Bake Light' flag is on.
```

I re-added the camera attributes .tres to the LightmapGI node. I was checking if that was the issue. In another large scene that I have, having the campera attributes to the LightmapGI node make the dynamic object to stop working with the probe


[Lightmap2.zip](https://github.com/godotengine/godot/files/11925570/Lightmap2.zip)

**Comment by Calinou** [comment]
I can open that MRP, but I get a bunch of messages when opening the scene:

```
 scene/resources/importer_mesh.cpp:1103 - Parameter "array_mesh_lightmap_unwrap_callback" is null.
  scene/resources/importer_mesh.cpp:1103 - Parameter "array_mesh_lightmap_unwrap_callback" is null.
```

This is likely why I can't bake lightmaps. I'm on v4.1.beta.custom_build [3aaa78709].

**Comment by Calinou** [comment]
I can confirm this on 4.0.3.stable (Linux, GeForce RTX 4090 with NVIDIA 535.54.03):

![image](https://github.com/godotengine/godot/assets/180032/44a64a65-aafb-49ff-9d3e-32b04ab2d217)

**Comment by gongpha** [comment]
weird. I still couldn't reproduce this in both versions (4.0.3 stable and 4.1 46424488edc341b65467ee7fd3ac423e4d49ad34) after the 3 retries, even though  I've done @Calinou 's instructions.
( Windows 11 x64, Vulkan, NVIDIA GeForce GTX 1650 536.23 )
![image](https://github.com/godotengine/godot/assets/13400398/4da3ded6-d7dd-4b78-a09a-ded4d2d5dd74)

**Comment by jcostello** [comment]
> weird. I still couldn't reproduce this in both versions (4.0.3 stable and 4.1 [4642448](https://github.com/godotengine/godot/commit/46424488edc341b65467ee7fd3ac423e4d49ad34)) after the 3 retries, even though I've done @Calinou 's instructions. ( Windows 11 x64, Vulkan, NVIDIA GeForce GTX 1650 536.23 ) ![image](https://user-images.githubusercontent.com/13400398/250303491-4da3ded6-d7dd-4b78-a09a-ded4d2d5dd74.png)

Try re-adding the lightmap node

**Comment by jcostello** [comment]
@gongpha Try reading the lightmap node

**Comment by gongpha** [comment]
@jcostello  here's its properties
![image](https://github.com/godotengine/godot/assets/13400398/f321b420-4882-45c2-bf32-0a2101378851)

**Comment by jcostello** [comment]
@gongpha I ment re-adding the LightmapGI (remove and add it again)

**Comment by gongpha** [comment]
Tried, doesn't work either

**Comment by jcostello** [comment]
The problem happens, it is a real issue, but IDK the right conditions. Maybe debugging the code?
Could be a Linux issue?

**Comment by atirut-w** [comment]
> Could be a Linux issue?

Invalid probes also appear on Windows, IIRC.

**Comment by jcostello** [comment]
> > Could be a Linux issue?
> 
> Invalid probes also appear on Windows, IIRC.

You've been able to reproduce it?

**Comment by atirut-w** [comment]
I've only been able to replicate it with plane meshes where probes are forced to overlap with geometries. I need someone else to confirm if this is the same problem.

**Comment by jcostello** [comment]
Do you have the exact steps so @gongpha can replicate

**Comment by atirut-w** [comment]
The only reliable way I know of to cause this issue is to force light probes to overlap with geometries. I can link you the model I used later today.

**Comment by jcostello** [comment]
@atirut-w I did that in my MRP but it doesnt work for @gongpha

**Comment by gongpha** [comment]
Finally can replicate this in 4.1.stable after creating a massive giant probe swarm (3240 probes). It's like a kind of randomness.

(the orange wireframes are the hiding cube)
![image](https://github.com/godotengine/godot/assets/13400398/28f0b1d3-08a4-46dc-bc24-da0141b91448)

![image](https://github.com/godotengine/godot/assets/13400398/d786d2e8-24e9-4201-a94e-cc03595b7c24)
[i_97045961_001.zip](https://github.com/godotengine/godot/files/12061020/i_97045961_001.zip)

**Comment by atirut-w** [comment]
Here's the model I used in my previous screenshot. It causes invalid probes at 16 subdivisions or higher: https://sketchfab.com/3d-models/1-55d66195dc6448e6a3fb7a30b4b063ff.

**Comment by jcostello** [comment]
https://github.com/godotengine/godot/pull/82068 seens to fix the issue

**Comment by expenses** [comment]
<img width="1539" height="627" alt="Image" src="https://github.com/user-attachments/assets/df8d9f17-19aa-4dc2-80b3-94f20e07bb04" />

I'm getting this when I try and bake with a 'Custom Sky' (a panoramic environment map that should be mostly grey). Possibly a new issue?

**Comment by Calinou** [comment]
@expenses Please open a new issue with a minimal reproduction project attached. See also https://github.com/godotengine/godot/issues/94406 which was fixed in 4.4.
